### PR TITLE
Improve loop to avoid race condition

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
### Issue

- [poo#20592](https://progress.opensuse.org/issues/20592)

### Verification run

- [copland#1071#step/evolution/5](http://copland.arch.suse.de/tests/1071#step/evolution/5)

### CODE REVIEW

File: tests/x11/evolution.pm [hunk 1]
```diff
@@ -20,11 +20,18 @@ sub run {
...
-    do {
-        assert_screen \@tags;
-        send_key 'alt-o'                                  if match_has_tag('evolution-preview-release');
-        assert_and_click 'evolution-default-client-agree' if match_has_tag('evolution-default-client-ask');
-    } until (match_has_tag('test-evolution-1'));
+    my $times = scalar(@tags) - 1;
+    assert_screen \@tags;
+
+    unless (match_has_tag('test-evolution-1')) {
+        for (my $i == 0; $i < $times; $i++) {
```
Comment (okurz) [inline tests/x11/evolution.pm]: `for (0 .. $times)`

**Comment by SergioAtSUSE** [comment]
New verification run: [copland#1098](http://copland.arch.suse.de/tests/1098) (running)

**Comment by okurz** [comment]
not really running, is your worker alive?

**Comment by SergioAtSUSE** [comment]
Updated

File: tests/x11/evolution.pm [hunk 2]
```diff
@@ -20,11 +20,18 @@ sub run {
...
-    do {
-        assert_screen \@tags;
-        send_key 'alt-o'                                  if match_has_tag('evolution-preview-release');
-        assert_and_click 'evolution-default-client-agree' if match_has_tag('evolution-default-client-ask');
-    } until (match_has_tag('test-evolution-1'));
+    my $times = scalar(@tags) - 1;
+    assert_screen \@tags;
+
+    unless (match_has_tag('test-evolution-1')) {
+        for (0 .. $times) {
```
Comment (rwx788) [inline tests/x11/evolution.pm]: The outer loop becomes kind of useless, as we do our stuff in for loop, when we are done we do assert, which will fail the test if there is no such tag. So it's as good as putting "unless" statement instead.

File: tests/x11/evolution.pm [hunk 3]
```diff
@@ -20,11 +20,18 @@ sub run {
...
+            assert_and_click 'evolution-default-client-agree' if match_has_tag('evolution-default-client-ask');
+            send_key 'alt-o'                                  if match_has_tag('evolution-preview-release');
```
Comment (rwx788) [inline tests/x11/evolution.pm]: I wonder if we can have else if here and spare one check of tag, if click is done.

File: tests/x11/evolution.pm [hunk 4]
```diff
@@ -20,11 +20,18 @@ sub run {
...
+            wait_still_screen;
```
Comment (rwx788) [inline tests/x11/evolution.pm]: Be aware that still screen default similarity level is rather low and test may still go further even some small part of the screen is changing.

File: tests/x11/evolution.pm (hunk 2 above)
Comment (SergioAtSUSE) [inline tests/x11/evolution.pm]: I don't understand what you mean. Which outer loop?

Comment (rwx788) [inline tests/x11/evolution.pm]: Oh, sorry I confused unless with until. My bad =)

**Comment by SergioAtSUSE** [comment]
Waiting for verification run

**Pull request merged**